### ISSUE TITLE
Component Options do not update when the state changes

### ISSUE THREAD

**Issue by wongCH** [harshzalavadiya/react-multi-select-component#438]
Describe the bug:
When options state is updated, the dropdown do not show the new value.

To Reproduce:

```
import React, { useState } from "react";
import MultiSelect from "react-multi-select-component";
import "./styles.css";
export default function App() {
  const [selected, setSelected] = useState([]);
  const [options, setOptions] = useState([
    { label: "Grapes 🍇", value: "grapes" }
  ]);
  const AddOption = () => {
    setOptions([ { label: "Mango 🥭", value: "mango" },
    { label: "Strawberry 🍓", value: "strawberry" }
  ]);
  };
  return (
    <div>
      <h1>Select Fruits</h1>
      <p>{JSON.stringify(options)}</p>
      <MultiSelect
        options={options}
        value={selected}
        onChange={setSelected}
        labelledBy={"Select"}
      />
      <button onClick={AddOption}> Add option </button>
    </div>
  );
}
```
 
  
 
This only happens from version 4.0.2 onwards. No issue with version 4.0.1

Sample Code using Version 4.0.3
https://codesandbox.io/s/react-kerl-bug-0hm06?file=/src/App.js

Desired behaviour:
When Option state changes, component should reflect the state just like in version 4.0.1

**Comment by harshzalavadiya** [comment]
@wongCH It looks like you are trying to print `options` instead of `selected`

it should be something like below
```js.diff
- <p>{JSON.stringify(options)}</p>
+ <p>{JSON.stringify(selected)}</p>
```

**Comment by wongCH** [comment]
Hi @harshzalavadiya ,

When page first load, the Option only have Grape:

<img width="867" alt="Before" src="https://user-images.githubusercontent.com/6981489/127817091-7104fb7f-55ac-4cda-af30-91f848fab79b.png">
**Picture above depicts the Option state that only contains "Grape".**


To simplified my scenario,i wrote a function to overwrite the Option to the following:
`  "[{"label":"Mango 🥭","value":"mango"},{"label":"Strawberry 🍓","value":"strawberry"}]" `

Unfortunately, the dropdown component do not reflect the new values after i tried to add into the Option state
<img width="926" alt="After" src="https://user-images.githubusercontent.com/6981489/127820066-bcba1285-2dc9-43e6-a9af-d479f37495c1.png">
**Picture above depicts the Option state  been replace with Mango and strawberry but the Dropdown component still show Grape.**

**Comment by wongCH** [comment]
@harshzalavadiya ,
Below is the picture when i change the version to 4.0.1, version 4.0.2 and version 4.0.3 dropdown do not populate Mango and Strawberry:

<img width="1082" alt="correct" src="https://user-images.githubusercontent.com/6981489/127820611-8d51bcd2-7f7c-4f59-93e0-66753aafc527.png">

**Comment by harshzalavadiya** [comment]
@wongCH my bad, yes you are right, I looked at different thing, I'll try to see and fix this

**Comment by harshzalavadiya** [comment]
@wongCH issue is fixed in latest release `v4.0.4+`

**Comment by wongCH** [comment]
@harshzalavadiya  thank you so much.
